src/components/invocable/InvocationModal.jsx

```diff
@@ -225,28 +230,44 @@ export default class InvocationModal extends Component {
       recordType,
     } = this.props;
 
+    const {
+      csid,
+      getMimeTypes,
+    } = this.props;
+
     const {
       invocationDescriptor,
     } = this.state;
 
-    // TODO: Use reports/mimetypes endpoint to get options, and [recordType]/mimetypes to check if
-    // a format picker should be shown instead of hardcoding (batch/mimetypes will return 404, so
-    // format picker should not be shown).
-
     if (recordType === 'report') {
+      let mimeList = [];
+
+      if (getMimeTypes) {
```

Comment:
Don't need this. The report data is already in the `data` prop. Also React render methods should not have side effects, so if you needed to make a request, this wouldn't be the place to do it.